app/controllers/alchemy/api/nodes_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class Api::NodesController < Api::BaseController
+    before_action :load_node
+
+    def move
+      target_parent_node = Node.find(params[:target_parent_id])
+      @node.move_to_child_with_index(target_parent_node, params[:new_position])
```

Comment:
Oh, just saw that we do not have any authorization here.

```suggestion
      authorize! :update, @node
      target_parent_node = Node.find(params[:target_parent_id])
      @node.move_to_child_with_index(target_parent_node, params[:new_position])
```